### DISCUSSION TITLE
Adding a String Array (Text) in Content-Builder

### DISCUSSION THREAD

**Discussion by asdutoit** [strapi/strapi#7340] [Legacy Topics]
I noticed it is not currently possible to add a field that can contain an array of strings using the Content Builder. 
Any idea if it will be supported in the near future and if so, is there a perhaps a date?

**Comment by derrickmehaffy** [comment]
Hello!

Currently there is no plan to support an array field type (yet) and the current recommendation is to use a `manyWay` relation which would function in the same basic concept (manyWay relation is the far right side of the Admin panels relation matrix).

In this way you create a model to store those strings (or any other meta-data like say "tags" as an example) that allow you to update them while in production without changing code, and you setup a manyWay relation on the model you wish to use them (say "blog posts"). Logically it also acts the same way as you would have 3 database tables (for those databases that require it, IE SQL):

- Blog Posts (for main content)
- Tags (storage of the strings you can attach)
- Intermediary table that joins the two (instead of sending just the strings, you send an array of existing tag IDs)

The downside is obviously the tags need to exist to attach them to the blog-post so there is an extra Step from the API however we are looking at a feature to "create relation data on demand" within the admin panel to handle that.

I will ping @Aurelsicoko to see if he can provide any information about that above feature and if we can add it to our [feature roadmap](https://portal.productboard.com/strapi/1-public-roadmap/tabs/3-planned)

**Reply by asdutoit** [reply]
Hi @derrickmehaffy , appreciate your response.
It is exactly what I am doing now, just feels a bit complicated to store simple strings.   

I am using MongoDB as a db and my idea was to store the array of strings as an array / embedded document.

However, thankfully there is a solution and I am appreciative and thankful for your work and love Strapi!  

Let me know if you plan to add it to your feature roadmap.

Regards

**Reply by derrickmehaffy** [reply]
The main reason why it hasn't been added as a feature yet is simply due to the fact that Strapi is agnostic to the backend database and the SQL databases don't have the option to just store an array (and have proper search capabilities on that json array), so in that case it's better to use a relation.

When we build feature we do our best to ensure it works in the same one to one way for all databases and don't design features that only work on one database and not the other 4.

**Reply by asdutoit** [reply]
Thanks @derrickmehaffy , that makes perfect sense.    I did end up building a separate relation which works perfectly fine.

**Comment by lauriejim** [comment]
Hello! In fact, you can use the JSON field to store this kind of information.
The JSON field can store a JSON object and Array.
<img width="1003" alt="Screen Shot 2020-08-07 at 16 33 05" src="https://user-images.githubusercontent.com/5716596/89656422-da42e800-d8cb-11ea-8b15-d1df1e1f6d3d.png">

**Reply by derrickmehaffy** [reply]
Just something to note here, with the JSON field, you can't currently filter on the content. So if you don't need to filter this option will work.

**Reply by asdutoit** [reply]
Hi @lauriejim and @derrickmehaffy ,  thank you very much.   I will give it a go.   Thankfully I don't currently have a requirement for filters.   😁
